UI/BuildDesignatorWnd.cpp

```diff
@@ -241,25 +244,109 @@ namespace {
+            production_time = building_type->ProductionTime(empire_id, candidate_object_id);
+
+            main_text += "\n\n" + UserString("PRODUCTION_WND_TOOLTIP_PROD_COST") + ": " + DoubleToString(total_cost, 3, false);
+            main_text += "\n" + UserString("PRODUCTION_WND_TOOLTIP_PROD_TIME") + ": " + boost::lexical_cast<std::string>(production_time);
+
+            // show build conditions
+            std::string EnqueueAndLocationConditionFailedText = EnqueueAndLocationConditionDescription(item.name, candidate_object_id, empire_id, true);
+            if (!EnqueueAndLocationConditionFailedText.empty())
+                if (TemporaryPtr<UniverseObject> location = GetUniverseObject(candidate_object_id)) {
+                        std::string failed_cond_loc = boost::io::str(FlexibleFormat(UserString("PRODUCTION_WND_TOOLTIP_FAILED_COND")) % location->Name());
+                        main_text += "\n\n" + failed_cond_loc + ":\n" + EnqueueAndLocationConditionFailedText;
+            }
+
+            // create tooltip
             boost::shared_ptr<GG::BrowseInfoWnd> browse_wnd(new IconTextBrowseWnd(
-                ClientUI::BuildingIcon(item.name), UserString(item.name),
-                EnqueueAndLocationConditionDescription(item.name, candidate_object_id, empire_id)));
+                ClientUI::BuildingIcon(item.name), title, main_text));
             return browse_wnd;
-        } else if (item.build_type == BT_SHIP) {
-            const ShipDesign* ship_design = GetShipDesign(item.design_id);
-            const std::string& name = (ship_design ? ship_design->Name() : UserString("SHIP_DESIGN"));
+        }
+
+        // production item is a ship
+        if (item.build_type == BT_SHIP) {
+            const ShipDesign* design = GetShipDesign(item.design_id);
+            if (!design) {
+                boost::shared_ptr<GG::BrowseInfoWnd> browse_wnd;
+                return browse_wnd;
+            }
+
+            // create title, description, production time and cost, hull type
+            title = design->Name(true);
+            main_text = design->Description(true);
+            total_cost = design->ProductionCost(empire_id, candidate_object_id);
+            production_time = design->ProductionTime(empire_id, candidate_object_id);
+
+            main_text += "\n\n" + UserString("PRODUCTION_WND_TOOLTIP_PROD_COST") + ": " + DoubleToString(total_cost, 3, false);
+            main_text += "\n" + UserString("PRODUCTION_WND_TOOLTIP_PROD_TIME") + ": " + boost::lexical_cast<std::string>(production_time);
+
+            main_text += "\n\n" + UserString("ENC_SHIP_HULL") + ": " + UserString(design->Hull());
+
+            // load ship parts, stack ship parts that are used multiple times
+            std::string ship_parts_formatted;
+            std::map<std::string, int> ship_part_names;
+            std::vector<std::string, std::allocator<std::string> > parts = design->Parts();
```

Comment:
No need to specify the allocator if you're using the default.
But it should be a const std::vector<std::string>& anyway, as that's what ShipDesign::Parts(void) returns.